x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -154,23 +154,29 @@ private RestResponse buildResponse(EsqlQueryResponse esqlResponse) throws IOExce
     }
 
     /**
-     * Log the execution time and query when handling an ES|QL response.
+     * Log internal server errors all the time and log queries if debug is enabled.
      */
     public ActionListener<EsqlQueryResponse> wrapWithLogging() {
+        ActionListener<EsqlQueryResponse> listener = ActionListener.wrap(this::onResponse, ex -> {
+            logOnFailure(LOGGER, ex);
+            onFailure(ex);
+        });
+        if (LOGGER.isDebugEnabled() == false) {
```

Comment:
should it check for log level more than DEBUG?

Reply:
That's what `isDebugEnabled` does. TRACE will log as well.